core/src/com/unciv/models/ruleset/unique/Unique.kt

```diff
@@ -298,6 +298,9 @@ open class UniqueMap() {
     
     fun hasUnique(uniqueType: UniqueType, state: StateForConditionals = StateForConditionals.EmptyState) =
         getUniques(uniqueType).any { it.conditionalsApply(state) && !it.isTimedTriggerable }
+
+    fun hasUnique(uniqueTag: String, state: StateForConditionals = StateForConditionals.EmptyState) =
```

Comment:
change to hasTagUnique

Reply:
For all of these, I'm not sure I understand the difference in naming convention between tags and types. It would imply to me that the typed version should also be renamed to "hasTypedUnique". And since what we're doing between each version of the function is fundamentally the same code (it's literally copy-pasted), there doesn't seem to be a good reason for the distinction

Reply:
Uniques are *known text strings that indicate code*, they are known.
Tag uniques are *arbitrary text strings* that are used for matching only.
If we're looking for a known UniqueType, that's a unique. If we're looking for arbitrary text, as in filters, that's a tag.
The naming here still feels *very wrong*, I'm willing to give it a go but if it confuses me in the future I'll change it to tags.

Reply:
From a technical standpoint, I feel like the distinction between tags and uniques is minor. Once we actually have the unique, how we manipulate it is functionally the same, and the only difference is that for performance and maintainability reasons, we typically only deal in UniqueTypes

Actually, there is a spot where one can build their own functioning unique via tags. As a ruins reward, do ``"[{This Unit} {non-[Does not upgrade from ruins-like effects]}] upgrades for free including special upgrades"``. No, we don't have a unique for this. And from a modder perspective, there's no reason to view the tag as if it's just a tag